modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/EitherTTest.kt

```diff
@@ -4,70 +4,72 @@ import arrow.Kind
+import arrow.core.extensions.const.eqK.eqK
+import arrow.core.extensions.either.eq.eq
+import arrow.core.extensions.eq
 import arrow.core.extensions.id.applicative.applicative
-import arrow.core.extensions.id.functor.functor
+import arrow.core.extensions.id.eqK.eqK
 import arrow.core.extensions.id.monad.monad
 import arrow.core.extensions.id.traverse.traverse
 import arrow.core.extensions.monoid
 import arrow.core.extensions.option.functor.functor
-import arrow.core.fix
-import arrow.core.value
 import arrow.fx.ForIO
 import arrow.fx.IO
-import arrow.fx.extensions.io.applicativeError.attempt
 import arrow.fx.extensions.io.concurrent.concurrent
 import arrow.fx.mtl.concurrent
-import arrow.fx.typeclasses.seconds
 import arrow.mtl.extensions.eithert.alternative.alternative
-import arrow.mtl.extensions.eithert.applicative.applicative
 import arrow.mtl.extensions.eithert.divisible.divisible
-import arrow.mtl.extensions.eithert.functor.functor
 import arrow.mtl.extensions.eithert.semigroupK.semigroupK
 import arrow.mtl.extensions.eithert.traverse.traverse
 import arrow.test.UnitSpec
+import arrow.test.generators.genK
 import arrow.test.laws.AlternativeLaws
 import arrow.test.laws.ConcurrentLaws
 import arrow.test.laws.DivisibleLaws
 import arrow.test.laws.SemigroupKLaws
 import arrow.test.laws.TraverseLaws
 import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
+import io.kotlintest.properties.Gen
 import io.kotlintest.properties.forAll
 
 class EitherTTest : UnitSpec() {
 
-  fun <E, A> EQ(): Eq<Kind<EitherTPartialOf<ForIO, E>, A>> = Eq { a, b ->
-    a.value().attempt().unsafeRunTimed(60.seconds) == b.value().attempt().unsafeRunTimed(60.seconds)
-  }
-
   init {
+    val idEQK: EqK<Kind<Kind<ForEitherT, ForId>, Int>> = EitherT.eqK(Id.eqK(), Int.eq())
+
+    val ioEQK: EqK<Kind<Kind<ForEitherT, ForIO>, String>> = EitherT.eqK(IO.eqK(), Eq.any())
+
+    val constEQK: EqK<Kind<Kind<ForEitherT, Kind<ForConst, Int>>, Int>> = EitherT.eqK(Const.eqK(Int.eq()), Int.eq())
 
     testLaws(
       DivisibleLaws.laws(
         EitherT.divisible<ConstPartialOf<Int>, Int>(Const.divisible(Int.monoid())),
-        { EitherT(it.const()) },
-        Eq { a, b -> a.value().value() == b.value().value() }
+        EitherT.genK(Const.genK(Gen.int()), Gen.int()),
+        constEQK
       ),
       AlternativeLaws.laws(
         EitherT.alternative(Id.monad(), Int.monoid()),
         { EitherT.just(Id.applicative(), it) },
         { i -> EitherT.just(Id.applicative(), { j: Int -> i + j }) },
-        Eq { a, b ->
-          a.value().fix() == b.value().fix()
-        }
+        idEQK
       ),
-      ConcurrentLaws.laws<EitherTPartialOf<ForIO, String>>(EitherT.concurrent(IO.concurrent()), EQ(), EQ(), EQ()),
-      TraverseLaws.laws(EitherT.traverse<ForId, Int>(Id.traverse()), EitherT.functor<ForId, Int>(Id.functor()), { EitherT(Id(Right(it))) }, Eq.any()),
+      ConcurrentLaws.laws<EitherTPartialOf<ForIO, String>>(EitherT.concurrent(IO.concurrent()), ioEQK),
```

Comment:
@1Jajen1 i having issues with failing tests now. i merged the latest master with @nomisRev changes. 
Now "Concurrent Laws: cancelable should run CancelToken on cancel" is failing on my machine for EitherTTest, WriterTTest, KleisliTest and OptionTTest.

i tried to generalize the EqK instances used in the test so prohably i have introduced a problem there. Would be great if one of you could have a look because i have a hard time understanding what the failing test is actually testing :-D

its easy reproducible by focussing on the test (`Law("f:Concurrent Laws: cancelable should run CancelToken on cancel")`) and running e.g. OptionTTest.

Reply:
looks like latest merge from master resolved the issue :-D